### PR TITLE
Test issues and landing

### PR DESCRIPTION
**Pull request by vladskakun**

### CODE REVIEW

**Comment by TanyaKalistruk** [comment]
@vladskakun Do we have any tests here?

File: pages/ModalPage.py [hunk 1]
```diff
@@ -12,6 +11,11 @@ class ModalPage(BaseWrapper):
     FORM_PASSWORD_INP_XPATH = f"{MODAL_DIALOG_XPATH}//input[@name='password']"
     FORM_REGISTER_PASSWORD_REPEAT_INP_XPATH = f"{MODAL_DIALOG_XPATH}//input[@name='RepeatPassword']"
     FORM_BTN_XPATH = f"{MODAL_DIALOG_XPATH}//span[text() = '{{}}']"
+    SIGN_IN_BUTTON_TEXT = "Sign In"
+    FORM_PAGE_LOGIN_TEXT = "Login"
+    FORM_PAGE_REGISTER_TEXT = "Register"
+    FORM_PAGE_BUTTON_SING_IN_TEXT = "Sign Up"
+    SUCCESS_PAGE_ALERT_TEXT = "div.alert-success"
```
Comment (TanyaKalistruk) [inline pages/ModalPage.py:18]: Is not it a locator?
Let's group text and locator constants

File: pages/ModalPage.py [hunk 2]
```diff
@@ -42,6 +46,9 @@ def click_button(self, name_button):
         element = self.find_element_by_xpath(self.FORM_BTN_XPATH.format(name_button))
         element.click()
 
+    def get_success_register_text(self):
```
Comment (TanyaKalistruk) [inline pages/ModalPage.py:49]: Docstring?)

File: pages/ModalPage.py [hunk 3]
```diff
@@ -52,4 +59,18 @@ def login(self, username, password):
...
         self.send_email_input(username)
         self.send_password_input(password)
-        self.click_button(config.landing_sign_in_button)
+        self.click_button(self.SIGN_IN_BUTTON_TEXT)
+
+    def registration(self, username, password):
+        """
+        Registration method as a scenario
+        actor - is a person with own permissions (admin or user).
+        Method of precondition for testing
```
Comment (TanyaKalistruk) [inline pages/ModalPage.py:68]: Actually, this method can be used not only as a precondition

File: pages/ModalPage.py [hunk 4]
```diff
@@ -52,4 +59,18 @@ def login(self, username, password):
...
+        :param username: username
+        :param password: pass phrase
+        """
+        self.click_page(self.FORM_PAGE_REGISTER_TEXT)
+        self.send_email_input(username)
+        self.send_password_input(password)
+        self.send_password_repeat_input(password)
+        self.click_button(self.FORM_PAGE_BUTTON_SING_IN_TEXT)
```
Comment (TanyaKalistruk) [inline pages/ModalPage.py:76]: Why don't we use ButtonElement here?
The same question about InputElements

File: tests/test_issue.py [hunk 5]
```diff
@@ -1,21 +1,71 @@
...
+CHECK_RESOLVE_TEXT = "Resolve"
 
 
-def test_issue_filter(app):
+def test_issue_datepicker(app):
     """
         Testing the issue filtering using DatePicker`s and check result.
     """
     expected_result = 2
+    # start test
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:19]: go to Event Express and login as admin

File: tests/test_issue.py [hunk 6]
```diff
@@ -1,21 +1,71 @@
...
+    app.landing.go_to_site()
+    app.landing.sign_up_btn.click_btn_by_css()
+    app.modal.login(config.ADMIN_EMAIL, config.ADMIN_PASS)
+    app.landing.find_event_btn.click_btn_by_css()
+    # go to home page
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:24]: probably issue page

File: tests/test_issue.py [hunk 7]
```diff
@@ -1,21 +1,71 @@
...
+    app.navigation.go_to_page(ISSUES_PAGE_TEXT)
+    # go to Issues page
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:26]: Our comments about actions should be before code with this action

File: tests/test_issue.py [hunk 8]
```diff
@@ -1,21 +1,71 @@
...
+    app.issues.click_issue_status_filter(CHECK_OPEN_TEXT)
+    app.issues.status_btns.click_btn_by_name(SEARCH_BUTTON_TEXT)
+    assert expected_result == app.issues.get_amount_of_issue_results(), \
+        "amount of issue results doesn`t same as expected"
+
+
+def test_issue_reset_btn(app):
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:53]: Docsting?

File: tests/test_issue.py [hunk 9]
```diff
@@ -1,21 +1,71 @@
...
+    app.issues.date_from_dtp.write_date_to_datepicker(21, 6, 2021)
+    app.issues.date_to_dtp.write_date_to_datepicker(15, 9, 2022)
+    app.issues.status_btns.click_btn_by_name(SEARCH_BUTTON_TEXT)
+    assert expected_result == app.issues.get_amount_of_issue_results(), \
+        "amount of issue results doesn`t same as expected"
+
+
+def test_issue_filter(app):
+    """
+        Testing the issue filtering using DatePicker`s and check result.
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:36]: Let's have docstrings for tests in the format
`Verify ...`

Comment (vladskakun) [reply tests/test_issue.py]: done

File: tests/test_issue.py (hunk 8 above)
Comment (vladskakun) [reply tests/test_issue.py]: done

File: tests/test_issue.py (hunk 7 above)
Comment (vladskakun) [reply tests/test_issue.py]: done

File: tests/test_issue.py (hunk 6 above)
Comment (vladskakun) [reply tests/test_issue.py]: yep, done

File: tests/test_issue.py (hunk 5 above)
Comment (vladskakun) [reply tests/test_issue.py]: Use allure(can be updated in next PR)

File: pages/ModalPage.py (hunk 4 above)
Comment (vladskakun) [reply pages/ModalPage.py]: done

File: pages/ModalPage.py (hunk 3 above)
Comment (vladskakun) [reply pages/ModalPage.py]: done

File: pages/ModalPage.py (hunk 2 above)
Comment (vladskakun) [reply pages/ModalPage.py]: done

File: pages/ModalPage.py (hunk 1 above)
Comment (vladskakun) [reply pages/ModalPage.py]: done)

File: pages/common/BaseWrapper.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    def find_element_by_css(self, locator, timeout=10):
+        """
+            Method for search element by css selector with wait
+        """
+        try:
+            WebDriverWait(self.driver, timeout).until(EC.presence_of_element_located((By.CSS_SELECTOR, locator)),
+                                                      message=f"Can't find element by locator {locator}")
+            return self.driver.find_element(By.CSS_SELECTOR, locator)
+        except:
+            allure.attach(self.driver.get_screenshot_as_png(), name="Screenshot",
```
Comment (TanyaKalistruk) [inline pages/common/BaseWrapper.py:27]: Do you have any ideas about the screenshot name?

File: pages/elements/InputElement.py [hunk 11]
```diff
@@ -14,8 +14,14 @@ def __init__(self, selector, driver):
 
     def send_data(self, string):
         """
-        Method for sending data to input field.
+        Method for sending data to input field by css.
         :param string: Variable string should contain text which we need to enter.
         """
         self.find_element_by_css(self.selector).send_keys(string)
 
+    def send_data_by_xpath(self, string):
```
Comment (TanyaKalistruk) [inline pages/elements/InputElement.py:22]: How about renaming the previous method for send_data_by_css?
The same question about methods in ButtonElement class

File: tests/test_issue.py [hunk 12]
```diff
@@ -1,21 +1,69 @@
+import allure
+import config
+
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:3]: Imports should be after docstring in the file

File: tests/test_issue.py [hunk 13]
```diff
@@ -1,21 +1,69 @@
...
-    assert expected_result == app.issues.get_amount_of_issue_results(), \
+    admin_setup.landing.find_event_btn.click_btn_by_css()
+    with allure.step("Click find event button and go to issue page"):
+        admin_setup.navigation.go_to_page(ISSUES_PAGE_TEXT)
+    with allure.step("Filling datepickers"):
+        admin_setup.issues.date_from_dtp.write_date_to_datepicker(21, 6, 2021)
+        admin_setup.issues.date_to_dtp.write_date_to_datepicker(15, 9, 2022)
+    with allure.step("Click search button"):
+        admin_setup.issues.status_btns.click_btn_by_name(SEARCH_BUTTON_TEXT)
+    with allure.step("Checking excepted result"):
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:28]: Checking amount of issues

File: tests/test_issue.py [hunk 14]
```diff
@@ -1,21 +1,69 @@
...
+        admin_setup.issues.status_btns.click_btn_by_name(SEARCH_BUTTON_TEXT)
+    assert expected_result == admin_setup.issues.get_amount_of_issue_results(), \
+        "amount of issue results doesn`t same as expected"
+
+
+def test_issue_reset_btn(admin_setup):
+    """
+        Verify that admin have ability clear input fields clicking by button 'RESET'.
+    """
+    ###
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:53]: What is it?

File: tests/test_landing.py [hunk 15]
```diff
@@ -11,23 +13,30 @@
 CHECK_RESOLVE_TEXT = "Resolve"
 
 
-def test_landing_login(app):
+@allure.title("Test login:")
+@allure.severity(Severity.BLOCKER)
```
Comment (TanyaKalistruk) [inline tests/test_landing.py:17]: Do you plan to add this for other tests?

Comment (vladskakun) [reply tests/test_landing.py:17]: Done, but I have trouble using allure...

File: tests/test_issue.py (hunk 14 above)
Comment (vladskakun) [reply tests/test_issue.py]: It`s flag for me) Sorry, I fix it

File: tests/test_issue.py (hunk 13 above)
Comment (vladskakun) [reply tests/test_issue.py]: done

File: tests/test_issue.py [hunk 16]
```diff
@@ -1,21 +1,69 @@
...
+    with allure.step("Click search button"):
+        admin_setup.issues.status_btns.click_btn_by_name(SEARCH_BUTTON_TEXT)
+    with allure.step("Checking excepted result"):
+        assert expected_result == admin_setup.issues.get_amount_of_issue_results(), \
+            "amount of issue results doesn`t same as expected"
+
+
+def test_issue_filter(admin_setup):
+    """
+        Verify that admin have ability filter issue using status CheckBox and check result.
```
[the remark this answers is not in the document]
Comment (vladskakun) [reply tests/test_issue.py]: Thanks, done

File: tests/test_issue.py (hunk 12 above)
Comment (vladskakun) [reply tests/test_issue.py]: done

File: pages/elements/InputElement.py (hunk 11 above)
Comment (vladskakun) [reply pages/elements/InputElement.py:22]: Oh, I thought about it. Thanks, done

File: config.py [hunk 17]
```diff
@@ -1,7 +1,7 @@
 """
     Test data for logging in us admin
```
Comment (nvoloshchuk19) [inline config.py:2]: typo `us`

File: config.py [hunk 18]
```diff
@@ -1,7 +1,7 @@
...
 """
-admin_email = "admin@gmail.com"
-admin_pass = "1qaz1qaz"
-landing_sign_in_button = "Sign In"
-base_url = "https://eventsexpress-test.azurewebsites.net/"
+ADMIN_EMAIL = "admin@gmail.com"
+ADMIN_PASS = "1qaz1qaz"
+LANDING_SIGN_IN_GUTTON = "Sign In"
```
Comment (nvoloshchuk19) [inline config.py:6]: typo `GUTTON`

File: pages/ModalPage.py [hunk 19]
```diff
@@ -1,55 +1,59 @@
...
         :param password: pass phrase
         """
-        self.send_email_input(username)
-        self.send_password_input(password)
-        self.click_button(config.landing_sign_in_button)
+        self.page_btns.click_btn_by_name_by_xpath(FORM_PAGE_REGISTER_TEXT)
+        self.email_inp.send_data_by_xpath(username)
+        self.pass_inp.send_data_by_xpath(password)
+        self.pass_repeat_inp.send_data_by_xpath(password)
+        #self.sign_in_up_clear_btns.click_btn_by_name_by_xpath(FORM_PAGE_BUTTON_SING_UP_TEXT)
```
Comment (nvoloshchuk19) [inline pages/ModalPage.py:59]: ?

Comment (vladskakun) [reply pages/ModalPage.py:59]: It`s ok)
Me and @TanyaKalistruk decided to comment out this line so as not to create new users now

File: config.py (hunk 18 above)
Comment (vladskakun) [reply config.py]: Oh, thanks, done

File: pages/ModalPage.py (hunk 19 above)
Comment (TanyaKalistruk) [reply pages/ModalPage.py:59]: @vladskakun we talked about skipping this test for now using pytest, not commenting this line :smile:

File: pages/Issues.py [hunk 20]
```diff
@@ -46,3 +46,9 @@ def get_amount_of_issue_results(self):
             Method for get amount of issues.
         """
         return len(self.find_elements(self.ISSUE_RESULTS_CSS))
+
+    def get_datepicker_text(self):
+        """
+            Method for get value of input.
```
Comment (TanyaKalistruk) [inline pages/Issues.py:52]:
```suggestion
            Returns value from datepicker field.
```

Comment (vladskakun) [reply pages/Issues.py]: Thanks, done

File: pages/common/BaseWrapper.py (hunk 10 above)
Comment (vladskakun) [reply pages/common/BaseWrapper.py]: done in fixture

File: pages/ModalPage.py (hunk 19 above)
Comment (vladskakun) [reply pages/ModalPage.py:59]: done

File: pages/Issues.py [hunk 21]
```diff
@@ -46,3 +46,9 @@ def get_amount_of_issue_results(self):
             Method for get amount of issues.
         """
         return len(self.find_elements(self.ISSUE_RESULTS_CSS))
+
+    def get_datepicker_text(self):
+        """
+            Returns value from datepicker field.
+        """
+        return self.find_element_by_css(self.DATAPICKER_FROM_CSS).text
```
Comment (TanyaKalistruk) [inline pages/Issues.py:54]: How about adding this method to DatePicker element?

File: tests/conftest.py [hunk 22]
```diff
@@ -1,20 +1,53 @@
...
     """
-    app.modal.login(config.admin_email, config.admin_pass)
+    app.go_to_site()
+    app.landing.sign_up_btn.click_btn_by_css()
+    app.modal.login(config.ADMIN_EMAIL, config.ADMIN_PASS)
+    return app
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item):
```
Comment (TanyaKalistruk) [inline tests/conftest.py:29]: How is it used?

File: tests/conftest.py [hunk 23]
```diff
@@ -1,20 +1,53 @@
...
+    Make screenshot on a test failure
+    """
+    yield
+    if request.node.rep_setup.failed:
+        make_screenshot(app.driver, request.function.__name__)
+    elif request.node.rep_call.failed:
+        make_screenshot(app.driver, request.function.__name__)
+
+
+def make_screenshot(driver, function_name):
```
Comment (TanyaKalistruk) [inline tests/conftest.py:50]: Docstring?)

File: tests/test_issue.py [hunk 24]
```diff
@@ -1,21 +1,77 @@
...
+SEARCH_BUTTON_TEXT = "SEARCH"
+RESET_BUTTON_TEXT = "RESET"
+CHECK_OPEN_TEXT = "Open"
+CHECK_IN_PROGRESS_TEXT = "In progress"
+CHECK_RESOLVE_TEXT = "Resolve"
+
+
+@allure.title("Test datepickers:")
+@allure.severity(Severity.NORMAL)
+def test_issue_datepicker(admin_setup):
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:20]: test_datepicker_on_issue_page

File: tests/test_issue.py [hunk 25]
```diff
@@ -1,21 +1,77 @@
...
+    with allure.step("Checking the amount of issues"):
+        assert expected_result == admin_setup.issues.get_amount_of_issue_results(), \
+            "amount of issue results doesn`t same as expected"
+
+
+@allure.title("Test checkbox filters:")
+@allure.severity(Severity.NORMAL)
+def test_issue_filter(admin_setup):
+    """
+        Verify that the admin has the ability to filter issues using status CheckBox and check results.
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:42]: Actually I suppose that `and check results` at the end is redundant)

File: tests/test_issue.py [hunk 26]
```diff
@@ -1,21 +1,77 @@
...
+    """
+    expected_result = 3
+    admin_setup.landing.find_event_btn.click_btn_by_css()
+    with allure.step("Click find event button and go to issue page"):
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:46]: Should not this allure step be before 45 line?

File: tests/test_issue.py [hunk 27]
```diff
@@ -1,21 +1,77 @@
...
+    with allure.step("Checking the amount of issues"):
+        assert expected_result == admin_setup.issues.get_amount_of_issue_results(), \
+            "amount of issue results doesn`t same as expected"
+
+
+@allure.title("Test 'Reset button':")
+@allure.severity(Severity.NORMAL)
+def test_issue_reset_btn(admin_setup):
+    """
+        Verify that admin have ability clear input fields clicking by button 'RESET'.
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:61]: Please, check all docstrings

File: tests/test_issue.py [hunk 28]
```diff
@@ -1,21 +1,77 @@
...
+    """
+    expected_result = ""
+    admin_setup.landing.find_event_btn.click_btn_by_css()
+    with allure.step("Click find event button and go to issue page"):
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:65]: The same question about allure step

File: tests/test_landing.py [hunk 29]
```diff
@@ -11,23 +14,33 @@
...
     expected_result = "Admin"
-    app.landing.go_to_site()
-    app.landing.sign_up_btn.click_btn_by_css()
-    app.modal.login(config.ADMIN_EMAIL, config.ADMIN_PASS)
-    app.landing.find_event_btn.click_btn_by_css()
-    assert expected_result == app.navigation.get_user_name(), \
-        "username results doesn`t same as expected"
+    with allure.step("Click find event button and go to home page"):
+        admin_setup.landing.find_event_btn.click_btn_by_css()
+    with allure.step("Checking excepted result"):
```
Comment (TanyaKalistruk) [inline tests/test_landing.py:26]: Verify username is as expected.

File: tests/test_landing.py [hunk 30]
```diff
@@ -11,23 +14,33 @@
...
-    app.landing.sign_up_btn.click_btn_by_css()
-    #app.modal.registration()
-    assert expected_result == app.modal.get_success_register_text(), \
-        "alert message doesn`t same as expected"
+    with allure.step("Go to site and click sign up button"):
+        app.landing.go_to_site()
+        app.landing.sign_up_btn.click_btn_by_css()
+    with allure.step("Go to 'Registration' page"):
+        app.modal.registration(config.ADMIN_EMAIL, config.ADMIN_PASS)
+    with allure.step("Checking excepted result"):
```
Comment (TanyaKalistruk) [inline tests/test_landing.py:44]: Verify registration was successful

Comment (vladskakun) [reply tests/test_landing.py:44]: Thank you, done

File: tests/test_landing.py (hunk 29 above)
Comment (vladskakun) [reply tests/test_landing.py]: Thank you, done

File: tests/test_issue.py (hunk 28 above)
Comment (vladskakun) [reply tests/test_issue.py:64]: done

File: tests/test_issue.py (hunk 25 above)
Comment (vladskakun) [reply tests/test_issue.py]: Ok, done

File: tests/test_issue.py [hunk 31]
```diff
@@ -1,21 +1,77 @@
...
+CHECK_IN_PROGRESS_TEXT = "In progress"
+CHECK_RESOLVE_TEXT = "Resolve"
+
+
+@allure.title("Test datepickers:")
+@allure.severity(Severity.NORMAL)
+def test_issue_datepicker(admin_setup):
     """
-        Testing the issue filtering using DatePicker`s and check result.
+        Verify that admin have ability filter issue using DatePicker`s and check result.
```
[the remark this answers is not in the document]
Comment (vladskakun) [reply tests/test_issue.py]: Thanks, done

File: tests/test_issue.py (hunk 24 above)
Comment (vladskakun) [reply tests/test_issue.py]: done

File: tests/conftest.py (hunk 23 above)
Comment (vladskakun) [reply tests/conftest.py:50]: done

File: tests/conftest.py (hunk 22 above)
Comment (vladskakun) [reply tests/conftest.py:29]: As far as I know, if you need to process pytest reports or access the environment executing tests, you can implement a hook that will be called during the creation of the "report" object.

File: pages/Issues.py (hunk 21 above)
Comment (vladskakun) [reply pages/Issues.py]: done

File: pages/ModalPage.py (hunk 19 above)
Comment (vladskakun) [reply pages/ModalPage.py]: My bad, done

File: pages/ModalPage.py [hunk 32]
```diff
@@ -1,55 +1,59 @@
...
+        self.pass_inp = InputElement(self.FORM_PASSWORD_INP_XPATH, driver)
+        self.pass_repeat_inp = InputElement(self.FORM_REGISTER_PASSWORD_REPEAT_INP_XPATH, driver)
+        self.sign_in_up_clear_btns = ButtonElements(self.FORM_BTN_XPATH, driver)
+        self.page_btns = ButtonElements(self.FORM_PAGE_XPATH, driver)
 
-    def click_page(self, page_title):
+    def get_success_register_text(self):
         """
-            Method for click on page depending on page_title value.
+            Method for return string that contains on the success register page, for check expected result.
```
Comment (TanyaKalistruk) [inline pages/ModalPage.py:33]:
```suggestion
            Method for getting the message about successful registration.
```

File: tests/conftest.py [hunk 33]
```diff
@@ -1,20 +1,56 @@
...
+    """
+        Method for making screenshot.
```
Comment (TanyaKalistruk) [inline tests/conftest.py:52]:
```suggestion
        Method for making a screenshot.
```

File: tests/test_issue.py (hunk 27 above)
Comment (TanyaKalistruk) [reply tests/test_issue.py:61]: I don't see it)

File: tests/test_landing.py [hunk 34]
```diff
@@ -11,23 +14,33 @@
 CHECK_RESOLVE_TEXT = "Resolve"
 
 
-def test_landing_login(app):
+@allure.title("Test login:")
+@allure.severity(Severity.BLOCKER)
+def test_landing_login(admin_setup):
     """
-        Testing the issue filtering using DatePicker`s and check result.
+        Verify that user have ability log in as a Admin.
```
Comment (TanyaKalistruk) [inline tests/test_landing.py:21]: user has the ability to login

File: tests/test_landing.py [hunk 35]
```diff
@@ -11,23 +14,33 @@
 CHECK_RESOLVE_TEXT = "Resolve"
 
 
-def test_landing_login(app):
+@allure.title("Test login:")
+@allure.severity(Severity.BLOCKER)
+def test_landing_login(admin_setup):
     """
-        Testing the issue filtering using DatePicker`s and check result.
+         Verify that user has the ability to login in as a Admin.
```
Comment (TanyaKalistruk) [inline tests/test_landing.py:21]: an admin)

File: tests/test_issue.py [hunk 36]
```diff
@@ -1,21 +1,77 @@
...
+    with allure.step("Verify that amount of issues equals expected quantity"):
+        assert expected_result == admin_setup.issues.get_amount_of_issue_results(), \
+            "amount of issue results doesn`t same as expected"
+
+
+@allure.title("Test 'Reset button':")
+@allure.severity(Severity.NORMAL)
+def test_issue_reset_btn(admin_setup):
+    """
+        Verify that admin has the ability clear input fields clicking by button 'RESET'.
```
Comment (TanyaKalistruk) [inline tests/test_issue.py:61]: ability to clear

**Review by TanyaKalistruk — APPROVED** [review]

**Pull request merged**